src/extension/actions.js

```diff
@@ -326,6 +326,55 @@ export async function getProfilePicture(_, { owner }) {
   return projects.find((p) => p.owner === owner)?.picture;
 }
 
+/**
+ * Tries to guess if a URL is an AEM site.
+ * @param {chrome.tabs.Tab} _ The tab
+ * @param {Object} message The message object
+ * @param {string} message.url The URL to check
+ * @returns {Promise<boolean>} True if the provided URL is an AEM site
+ */
+export async function guessAEMSite(_, { url }) {
+  try {
+    const resp = await fetch(url, { redirect: 'manual' }); // don't allow redirects
+    if (resp.ok) {
+      const mimeType = resp.headers.get('content-type');
+      if (mimeType.startsWith('text/html')) {
+        // payload must match pipeline output
+        const payload = await resp.text();
+        const [type, html, head, titl, link, meta] = payload.substring(0, 400).split('\n');
+        return type === '<!DOCTYPE html>'
+          && html.startsWith('<html')
+          && head === '  <head>'
+          && titl.startsWith('    <title>')
+          && link.startsWith('    <link rel="canonical"')
+          && meta.startsWith('    <meta');
+      } else if (mimeType.startsWith('application/json')) {
+        // payload must match pipeline output
+        try {
+          const payload = await resp.json();
+          return !!payload[':type'];
+        } catch (e) {
+          // invalid json
+          return false;
+        }
+      } else {
+        // unable to guess. assume AEM
+        return true;
+      }
+    } else {
+      if (resp.status === 404 || resp.status >= 500) {
+        // unexpected not found or server error, assume incorrect prod setup
+        return false;
+      }
+      // otherwise assume AEM (e.g. protected)
+      return true;
```

Comment:
I did try a couple of production hosts and seemed to hit this fallback a few times. For example ifdesign.com or cnn.com. I think it might be related to redirects.

Reply:
hmm, but wouldn't `30x` be an `ok` respone status?